vyper/semantics/validation/annotation.py

```diff
@@ -172,7 +178,7 @@ def visit_List(self, node, type_):
             return
         if type_ is None:
             type_ = get_possible_types_from_node(node)
-            if len(type_) == 1:
+            if len(type_) > 1:
```

Comment:
Reason for this change?

Reply:
Literal lists can have multiple possible types (dynamic or static array), the typechecker has to resolve which one it is